File: 2- Calculadora/cs/CalculadoraCS/CalculadoraCS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace CalculadoraCS
{
	public partial class CalculadoraCS : Form
	{

		double primerNumero = 0;
		double segundoNumero = 0;
		bool punto = false;
		bool repeticion = false;

		int estado = 0;

		string operacionRealizar = " ";


		public CalculadoraCS()
		{
			InitializeComponent();
		}

		private void btn0_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
				{
				txtDisplay.Text = txtDisplay.Text + "0";
				}
			else
			{
				txtDisplay.Text = "0";
				estado = 0;
			}
		}

		private void btn1_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "1";
			}
			else
			{
				txtDisplay.Text = "1";
				estado = 0;
			}
		}

		private void btn2_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "2";
			}
			else
			{
				txtDisplay.Text = "2";
				estado = 0;
			}
		}

		private void btn3_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "3";
			}
			else
			{
				txtDisplay.Text = "3";
				estado = 0;
			}
		}

		private void btn4_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "4";
			}
			else
			{
				txtDisplay.Text = "4";
				estado = 0;
			}
		}

		private void btn5_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "5";
			}
			else
			{
				txtDisplay.Text = "5";
				estado = 0;
			}
		}

		private void btn6_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "6";
			}
			else
			{
				txtDisplay.Text = "6";
				estado = 0;
			}
		}

		private void btn7_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "7";
			}
			else
			{
				txtDisplay.Text = "7";
				estado = 0;
			}
		}

		private void btn8_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "8";
			}
			else
			{
				txtDisplay.Text = "8";
				estado = 0;
			}
		}

		private void btn9_Click(object sender, EventArgs e)
		{
			repeticion = false;

			if (estado == 0)
			{
				txtDisplay.Text = txtDisplay.Text + "9";
			}
			else
			{
				txtDisplay.Text = "9";
				estado = 0;
			}
		}

		private void btnPunto_Click(object sender, EventArgs e)
		{

			if (estado == 0)
			{
				if (punto == false)
				{
					txtDisplay.Text = txtDisplay.Text + ".";
					punto = true;
					repeticion = false;
				}
				else if (punto == true)
				{
					punto = true;
				}
			}
			else
			{
				if (punto == false)
				{
					txtDisplay.Text = ".";
					estado = 0;
					punto = true;
					repeticion = false;
				}
				else if (punto == true)
				{
					punto = true;
				}
			}
		}













		private void btnSumar_Click(object sender, EventArgs e)
		{

			if (repeticion == false)
			{

				repeticion = true;

				if(operacionRealizar == " ")
				{
					primerNumero = Convert.ToDouble(txtDisplay.Text);
					txtDisplay.Text = " ";
					operacionRealizar = "Suma";
					punto = false;
				}
				else if (operacionRealizar == "Suma")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero + segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Suma";
					punto = false;
				}
				else if (operacionRealizar == "Resta")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero - segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Suma";
					punto = false;
				}
				else if (operacionRealizar == "Multiplicacion")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero * segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Suma";
					punto = false;
				}
				else if (operacionRealizar == "Division")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero / segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Suma";
					punto = false;
				}
				else if (operacionRealizar == "Pow")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = Math.Pow((primerNumero),(segundoNumero));
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Suma";
					punto = false;
				}

			}

		}




		private void btnRestar_Click(object sender, EventArgs e)
		{
			if (repeticion == false)
			{

				repeticion = true;

				if (operacionRealizar == " ")
				{
					primerNumero = Convert.ToDouble(txtDisplay.Text);
					txtDisplay.Text = " ";
					operacionRealizar = "Resta";
					punto = false;
				}
				else if (operacionRealizar == "Suma")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero + segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Resta";
					punto = false;
				}
				else if (operacionRealizar == "Resta")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero - segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Resta";
					punto = false;
				}
				else if (operacionRealizar == "Multiplicacion")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero * segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Resta";
					punto = false;
				}
				else if (operacionRealizar == "Division")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero / segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Resta";
					punto = false;
				}
				else if (operacionRealizar == "Pow")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = Math.Pow((primerNumero), (segundoNumero));
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Resta";
					punto = false;
				}

			}
		}

		private void btnMultiplicar_Click(object sender, EventArgs e)
		{
			if (repeticion == false)
			{

				repeticion = true;

				if (operacionRealizar == " ")
				{
					primerNumero = Convert.ToDouble(txtDisplay.Text);
					txtDisplay.Text = " ";
					operacionRealizar = "Multiplicacion";
					punto = false;
				}
				else if (operacionRealizar == "Suma")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero + segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Multiplicacion";
					punto = false;
				}
				else if (operacionRealizar == "Resta")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero - segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Multiplicacion";
					punto = false;
				}
				else if (operacionRealizar == "Multiplicacion")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero * segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Multiplicacion";
					punto = false;
				}
				else if (operacionRealizar == "Division")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero / segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Multiplicacion";
					punto = false;
				}
				else if (operacionRealizar == "Pow")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = Math.Pow((primerNumero), (segundoNumero));
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Multiplicacion";
					punto = false;
				}

			}
		}

		private void btnDividir_Click(object sender, EventArgs e)
		{
			if (repeticion == false)
			{

				repeticion = true;

				if (operacionRealizar == " ")
				{
					primerNumero = Convert.ToDouble(txtDisplay.Text);
					txtDisplay.Text = " ";
					operacionRealizar = "Division";
					punto = false;
				}
				else if (operacionRealizar == "Suma")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero + segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Division";
					punto = false;
				}
				else if (operacionRealizar == "Resta")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero - segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Division";
					punto = false;
				}
				else if (operacionRealizar == "Multiplicacion")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero * segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Division";
					punto = false;
				}
				else if (operacionRealizar == "Division")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero / segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Division";
					punto = false;
				}
				else if (operacionRealizar == "Pow")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = Math.Pow((primerNumero), (segundoNumero));
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Division";
					punto = false;
				}

			}
		}

		private void btnPow_Click(object sender, EventArgs e)
		{
			if (repeticion == false)
			{

				repeticion = true;

				if (operacionRealizar == " ")
				{
					primerNumero = Convert.ToDouble(txtDisplay.Text);
					txtDisplay.Text = " ";
					operacionRealizar = "Pow";
					punto = false;
				}
				else if (operacionRealizar == "Suma")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero + segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Pow";
					punto = false;
				}
				else if (operacionRealizar == "Resta")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero - segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Pow";
					punto = false;
				}
				else if (operacionRealizar == "Multiplicacion")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero * segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Pow";
					punto = false;
				}
				else if (operacionRealizar == "Division")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = primerNumero / segundoNumero;
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Pow";
					punto = false;
				}
				else if (operacionRealizar == "Pow")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);
					primerNumero = Math.Pow((primerNumero), (segundoNumero));
					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = "Pow";
					punto = false;
				}

			}
			else
			{
				txtDisplay.Text = "";
			}
		}

		private void btnIgual_Click(object sender, EventArgs e)
		{

			if (txtDisplay.Text != "")
			{

				if(operacionRealizar == " ")
				{
					primerNumero = Convert.ToDouble(txtDisplay.Text);
					txtDisplay.Text = Convert.ToString(primerNumero);
					operacionRealizar = " ";
					punto = false;
				}
				else if (operacionRealizar == "Suma")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);

					primerNumero = primerNumero + segundoNumero;

					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = " ";
					punto = false;
				}
				else if (operacionRealizar == "Resta")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);

					primerNumero = primerNumero - segundoNumero;

					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = " ";
					punto = false;
				}
				else if (operacionRealizar == "Multiplicacion")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);

					primerNumero = primerNumero * segundoNumero;

					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = " ";
					punto = false;
				}
				else if (operacionRealizar == "Division")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);

					primerNumero = primerNumero / segundoNumero;

					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = " ";
					punto = false;
				}
				else if (operacionRealizar == "Pow")
				{
					segundoNumero = Convert.ToDouble(txtDisplay.Text);

					primerNumero = Math.Pow(primerNumero,segundoNumero);

					txtDisplay.Text = Convert.ToString(primerNumero);
					estado = 1;
					operacionRealizar = " ";
					punto = false;
				}

			}

		}

		private void btnClear_Click(object sender, EventArgs e)
		{
			txtDisplay.Text = " ";

			operacionRealizar = " ";

			punto = false;

			estado = 0;

			primerNumero = 0;

			segundoNumero = 0;
		}

		private void btnNeg_Click(object sender, EventArgs e)
		{
			if (txtDisplay.Text != ""){
				txtDisplay.Text = Convert.ToString(Convert.ToInt32(txtDisplay.Text) * (-1));
			}

			
		}
	}
}

File: 2- Calculadora/vb/Calculadora53/Calculadora53/Form1.vb
Public Class Calculadora

    Dim primerNumero As Double = 0
    Dim segundoNumero As Double = 0
    Dim punto As Boolean = False
    Dim repeticion As Boolean = False

    Dim estado As Integer = 0

    Dim operacionRealizar As String = " "





    Private Sub btn0_Click(sender As Object, e As EventArgs) Handles btn0.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "0"

        Else

            txtDisplay.Text = "0"
            estado = 0

        End If

    End Sub

    Private Sub btn1_Click(sender As Object, e As EventArgs) Handles btn1.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "1"

        Else

            txtDisplay.Text = "1"
            estado = 0

        End If

    End Sub

    Private Sub btn2_Click(sender As Object, e As EventArgs) Handles btn2.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "2"

        Else

            txtDisplay.Text = "2"
            estado = 0

        End If

    End Sub

    Private Sub btn3_Click(sender As Object, e As EventArgs) Handles btn3.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "3"

        Else

            txtDisplay.Text = "3"
            estado = 0

        End If

    End Sub

    Private Sub btn4_Click(sender As Object, e As EventArgs) Handles btn4.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "4"

        Else

            txtDisplay.Text = "4"
            estado = 0

        End If

    End Sub

    Private Sub btn5_Click(sender As Object, e As EventArgs) Handles btn5.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "5"

        Else

            txtDisplay.Text = "5"
            estado = 0

        End If

    End Sub

    Private Sub btn6_Click(sender As Object, e As EventArgs) Handles btn6.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "6"

        Else

            txtDisplay.Text = "6"
            estado = 0

        End If

    End Sub

    Private Sub btn7_Click(sender As Object, e As EventArgs) Handles btn7.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "7"

        Else

            txtDisplay.Text = "7"
            estado = 0

        End If

    End Sub

    Private Sub btn8_Click(sender As Object, e As EventArgs) Handles btn8.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "8"

        Else

            txtDisplay.Text = "8"
            estado = 0

        End If

    End Sub

    Private Sub btn9_Click(sender As Object, e As EventArgs) Handles btn9.Click

        repeticion = False
        If estado = 0 Then

            txtDisplay.Text = txtDisplay.Text + "9"

        Else

            txtDisplay.Text = "9"
            estado = 0

        End If

    End Sub

    Private Sub btnPunto_Click(sender As Object, e As EventArgs) Handles btnPunto.Click

        If estado = 0 Then
            If punto = False Then
                txtDisplay.Text = txtDisplay.Text + "."
                punto = True
                repeticion = False
            ElseIf punto = True Then
                punto = True
            End If
        Else

            If punto = False Then
                txtDisplay.Text = "."
                estado = 0
                punto = True
                repeticion = False
            ElseIf punto = True Then
                punto = True
            End If




        End If

    End Sub







    Private Sub btnSumar_Click(sender As Object, e As EventArgs) Handles btnSumar.Click

        If repeticion = False Then



            If operacionRealizar = " " Then

                primerNumero = Val(txtDisplay.Text)
                txtDisplay.Text = " "
                operacionRealizar = "Suma"
                punto = False


            ElseIf operacionRealizar = "Suma" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero + segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Suma"
                punto = False



            ElseIf operacionRealizar = "Resta" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero - segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Suma"
                punto = False


            ElseIf operacionRealizar = "Multiplicacion" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero * segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Suma"
                punto = False


            ElseIf operacionRealizar = "Division" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero / segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Suma"
                punto = False

            ElseIf operacionRealizar = "Pow" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero ^ segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Suma"
                punto = False

            End If

            repeticion = True

        End If

    End Sub

    Private Sub btnRestar_Click(sender As Object, e As EventArgs) Handles btnRestar.Click

        If repeticion = False Then

            If operacionRealizar = " " Then

            primerNumero = Val(txtDisplay.Text)
            txtDisplay.Text = " "
            operacionRealizar = "Resta"
            punto = False


        ElseIf operacionRealizar = "Suma" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero + segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Resta"
            punto = False



        ElseIf operacionRealizar = "Resta" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero - segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Resta"
            punto = False


        ElseIf operacionRealizar = "Multiplicacion" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero * segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Resta"
            punto = False


        ElseIf operacionRealizar = "Division" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero / segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Resta"
            punto = False


        ElseIf operacionRealizar = "Pow" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero ^ segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Resta"
            punto = False

        End If

            repeticion = True

        End If



    End Sub

    Private Sub btnMultiplicar_Click(sender As Object, e As EventArgs) Handles btnMultiplicar.Click

        If repeticion = False Then

            If operacionRealizar = " " Then

            primerNumero = Val(txtDisplay.Text)
            txtDisplay.Text = " "
            operacionRealizar = "Multiplicacion"
            punto = False


        ElseIf operacionRealizar = "Suma" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero + segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Multiplicacion"
            punto = False



        ElseIf operacionRealizar = "Resta" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero - segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Multiplicacion"
            punto = False


        ElseIf operacionRealizar = "Multiplicacion" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero * segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Multiplicacion"
            punto = False


        ElseIf operacionRealizar = "Division" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero / segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Multiplicacion"
            punto = False

        ElseIf operacionRealizar = "Pow" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero ^ segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Multiplicacion"
            punto = False

        End If


            repeticion = True

        End If

    End Sub

    Private Sub btnDividir_Click(sender As Object, e As EventArgs) Handles btnDividir.Click

        If repeticion = False Then

            If operacionRealizar = " " Then

            primerNumero = Val(txtDisplay.Text)
            txtDisplay.Text = " "
            operacionRealizar = "Division"
            punto = False


        ElseIf operacionRealizar = "Suma" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero + segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Division"
            punto = False



        ElseIf operacionRealizar = "Resta" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero - segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Division"
            punto = False


        ElseIf operacionRealizar = "Multiplicacion" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero * segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Division"
            punto = False


        ElseIf operacionRealizar = "Division" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero / segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Division"
            punto = False


        ElseIf operacionRealizar = "Pow" Then

            segundoNumero = Val(txtDisplay.Text)

            primerNumero = primerNumero ^ segundoNumero

            txtDisplay.Text = primerNumero
            estado = 1
            operacionRealizar = "Division"
            punto = False

        End If

            repeticion = True

        End If

    End Sub


    Private Sub btnPow_Click(sender As Object, e As EventArgs) Handles btnPow.Click

        If repeticion = False Then

            If operacionRealizar = " " Then

                primerNumero = Val(txtDisplay.Text)
                txtDisplay.Text = " "
                operacionRealizar = "Pow"
                punto = False


            ElseIf operacionRealizar = "Suma" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero + segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Pow"
                punto = False



            ElseIf operacionRealizar = "Resta" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero - segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Pow"
                punto = False


            ElseIf operacionRealizar = "Multiplicacion" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero * segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Pow"
                punto = False


            ElseIf operacionRealizar = "Division" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero / segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Pow"
                punto = False


            ElseIf operacionRealizar = "Pow" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero ^ segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = "Pow"
                punto = False

            End If

            repeticion = True

        End If
    End Sub





    Private Sub btnIgual_Click(sender As Object, e As EventArgs) Handles btnIgual.Click

        If txtDisplay.Text IsNot "" Then

            If operacionRealizar = " " Then

                primerNumero = Val(txtDisplay.Text)
                txtDisplay.Text = primerNumero
                operacionRealizar = " "
                punto = False

            ElseIf operacionRealizar = "Suma" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero + segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = " "
                punto = False



            ElseIf operacionRealizar = "Resta" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero - segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = " "
                punto = False


            ElseIf operacionRealizar = "Multiplicacion" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero * segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = " "
                punto = False


            ElseIf operacionRealizar = "Division" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero / segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = " "
                punto = False

            ElseIf operacionRealizar = "Pow" Then

                segundoNumero = Val(txtDisplay.Text)

                primerNumero = primerNumero ^ segundoNumero

                txtDisplay.Text = primerNumero
                estado = 1
                operacionRealizar = " "
                punto = False

            End If
        End If

    End Sub

    Private Sub btnClear_Click(sender As Object, e As EventArgs) Handles btnClear.Click

        txtDisplay.Text = " "
        operacionRealizar = " "
        punto = False
        estado = 0
        primerNumero = 0
        segundoNumero = 0

    End Sub

    Private Sub btnNeg_Click(sender As Object, e As EventArgs) Handles btnNeg.Click
        If txtDisplay.Text IsNot "" Then
            txtDisplay.Text = txtDisplay.Text * (-1)
        End If

    End Sub
End Class
